lib/gaffe/errors_controller_resolver.rb

```diff
@@ -30,7 +30,11 @@ def resolved_controller
   private
 
     def request_controller(controllers)
-      controllers.find { |pattern, _| @env['REQUEST_URI'] =~ pattern }.try(:last)
+      controllers.find do |pattern, _|
+        relative_url = @env['REQUEST_URI']
+        absolute_url = @env['HTTP_HOST'] + relative_url
+        !(relative_url =~ pattern).nil? || !(absolute_url =~ pattern).nil?
+      end.try(:last)
```

Comment:
I like the idea but could you fix these two things:

1. The condition is very difficult to read/understand at first glance. Can you rewrite it to be more verbose?
2. I don’t like `end.<method>`. The result of `#find` whould be stored in a variable and we should call `try(:last)` on it.